src/basic/conf-files.c

```diff
@@ -372,3 +374,87 @@ int conf_files_list_dropins(
 
         return conf_files_list_strv(ret, ".conf", root, 0, (const char* const*) dropin_dirs);
 }
+
+/**
+ * Open and read a config file.
+ *
+ * The <fn> argument may be:
+ * - '-', meaning stdin.
+ * - a non-absolute path. In this case <config_dirs> are searched.
+ * - an absolute path. In this case <fn> is opened directly.
+ */
+int conf_file_read(
+                const char *root,
+                const char **config_dirs,
```

Comment:
Maybe use `char * const *`? `const char **` is incompatible with `char **`, which makes it hard to use and we generally want to avoid it.